Doc/library/multiprocessing.shared_memory.rst

```diff
@@ -0,0 +1,381 @@
+   a string.  If ``None`` is supplied for the name, a new shared memory
+   block with a novel name will be created without needing to also specify
+   ``flags``.
+
+   *flags* is set to ``None`` when attempting to attach to an existing shared
+   memory block by its unique name but if no existing block has that name, an
+   exception will be raised.  To request the creation of a new shared
+   memory block, set to ``O_CREX``.  To request the optional creation of a
+   new shared memory block or attach to an existing one by the same name,
+   set to ``O_CREAT``.
+
+   *mode* controls user/group/all-based read/write permissions on the
+   shared memory block.  Its specification is not enforceable on all platforms.
+
+   *size* specifies the number of bytes requested for a shared memory block.
+   Because some platforms choose to allocate chunks of memory based upon
+   that platform's memory page size, the exact size of the shared memory
+   block may be larger or equal to the size requested.  When attaching to an
+   existing shared memory block, set to ``0`` (which is the default).
+
+   *read_only* controls whether a shared memory block is to be available
+   for only reading or for both reading and writing.  Its specification is
+   not enforceable on all platforms.
+
+   .. method:: close()
+
+      Closes access to the shared memory from this instance.  In order to
+      ensure proper cleanup of resources, all instances should call
+      ``close()`` once the instance is no longer needed.  Note that calling
+      ``close()`` does not cause the shared memory block itself to be
+      destroyed.
+
+   .. method:: unlink()
+
+      Requests that the underlying shared memory block be destroyed.  In
+      order to ensure proper cleanup of resources, ``unlink()`` should be
+      called once (and only once) across all processes which have need
+      for the shared memory block.  After requesting its destruction, a
+      shared memory block may or may not be immediately destroyed and
+      this behavior may differ across platforms.  Attempts to access data
+      inside the shared memory block after ``unlink()`` has been called may
+      result in memory access errors.  Note: the last process relinquishing
+      its hold on a shared memory block may call ``unlink()`` and
+      :meth:`close()` in either order.
+
+   .. attribute:: buf
+
+      A memoryview of contents of the shared memory block.
+
+   .. attribute:: name
+
+      Read-only access to the unique name of the shared memory block.
+
+   .. attribute:: mode
+
+      Read-only access to access permissions mode of the shared memory block.
+
+   .. attribute:: size
+
+      Read-only access to size in bytes of the shared memory block.
+
+
+The following example demonstrates low-level use of :class:`SharedMemory`
+instances::
+
+   >>> from multiprocessing import shared_memory
+   >>> shm_a = shared_memory.SharedMemory(None, size=10)
+   >>> type(shm_b.buf)
```

Comment:
Where do you define `shm_b?`

Reply:
Great spotting that missed detail!  Fixed in commit da7731db412a94d9f06a99500b31c9147e601e3e.